### PR TITLE
Freeze Spec Bug Fixes

### PR DESCRIPTION
**Pull request by terencechain**
This PR addresses the 3 PRs under `bugs` for the SubZero v0.8.0 tag release:
https://github.com/ethereum/eth2.0-specs/pull/1220
https://github.com/ethereum/eth2.0-specs/pull/1248
https://github.com/ethereum/eth2.0-specs/pull/1257

This wraps up the `bugs` fixes section for SubZero tag

### CODE REVIEW

File: beacon-chain/core/blocks/block_operations.go [hunk 1]
```diff
@@ -929,9 +929,12 @@ func verifyTransfer(beaconState *pb.BeaconState, transfer *pb.Transfer, verifySi
 	if transfer.Amount > maxVal {
 		maxVal = transfer.Amount
 	}
+	if transfer.Amount+transfer.Fee > maxVal {
+		maxVal = transfer.Amount + transfer.Fee
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/block_operations.go]: I don't feel good about this. Where's the test?

Why would we allow sender to initiate a transfer that exceeds the max val anyway? Shouldn't this be rejected?

Comment (terencechain) [reply beacon-chain/core/blocks/block_operations.go]: We dont allow a sender to initiate a transfer that exceeds maxVal, see L938 that's where we assert

this maxVal variable is used to check for over flow condition, we are not using it anywhere else

Comment (prestonvanloon) [reply beacon-chain/core/blocks/block_operations.go]: Can you add a test for this check specifically?

Comment (terencechain) [reply beacon-chain/core/blocks/block_operations.go]: yep, added!

**Pull request merged**